src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -1681,6 +1302,463 @@ DebugIpStatusCollector::populateILAResults(boost::property_tree::ptree &_pt)
+      const auto col2 = std::max(portNameMaxStrLen[AXI_STREAM_PROTOCOL_CHECKER], strlen("AXI Portname"));
+
+      boost::format header("  %-"+std::to_string(col1)+"s %-"+std::to_string(col2)+"s  %-16s  %-16s  %-16s");
+      _output << "\n"
+              << header % "CU Name" % "AXI Portname" % "Overall Status" % "Snapshot" % "Current"
+              << std::endl;
+
+      boost::format valueFormat("  %-"+std::to_string(col1)+"s %-"+std::to_string(col2)+"s  %-16x  %-16x  %-16x");
+      for(auto& ip : spc_pt) {
+        const boost::property_tree::ptree& entry = ip.second;
+        _output << valueFormat
+                     % entry.get<std::string>("cu_name") % entry.get<std::string>("axi_port")
+                     % entry.get<unsigned int>("pc_asserted") 
+                     % entry.get<unsigned int>("snapshot_pc") 
+                     % entry.get<unsigned int>("current_pc")
+                << std::endl;
+      }
+    }
+  }
+  catch( std::exception const& e) {
+    _output << "\nERROR: " <<  e.what() << std::endl;
+  }
+  _output << std::endl;
+}
+
+void
+reportILA(std::ostream& _output, const boost::property_tree::ptree& _pt, bool _gen_not_found_info)
+{
+  boost::optional<const boost::property_tree::ptree&> child = _pt.get_child_optional("Integrated Logic Analyzer");
+  if(boost::none == child) {
+    if(true == _gen_not_found_info) {
+      _output << "\nINFO: Element filter for ILA enabled but currently loaded xclbin does not have any ILA. So, ILA status report will NOT be generated." 
+              << std::endl;
+    }
+    return;
+  }
+  const boost::property_tree::ptree& ila_pt = child.get();
+  _output << boost::format("\nIntegrated Logic Analyzer\n  %s\n  Found : %llu") 
+              % ila_pt.get<std::string>("description") % ila_pt.get<uint64_t>("count")
+          << std::endl;
+}
+
+void
+reportAccelDeadlock(std::ostream& _output, const boost::property_tree::ptree& _pt, bool _gen_not_found_info)
+{
+  boost::optional<const boost::property_tree::ptree&> child = _pt.get_child_optional("accel_deadlock_detector_status");
+  if(boost::none == child) {
+    if(true == _gen_not_found_info) {
+      _output << "\nINFO: Element filter for Accelerator Deadlock Detector enabled but currently loaded xclbin does not have any Accelerator Deadlock Detector. So, Accelerator Deadlock Detector status report will NOT be generated." 
+              << std::endl;
+    }
+    return;
+  }
+  const boost::property_tree::ptree& accel_deadlock_pt = child.get();
+  _output << "\nAccelerator Deadlock Detector IP status :" 
+          << ((0 == accel_deadlock_pt.get<uint64_t>("is_deadlocked")) ? " No " : " ")
+          << "deadlock detected." << std::endl;
+}
+
+void 
+processElementFilter(bool *debugIpOpt, const std::vector<std::string> & _elementsFilter)
+{
+  // reset debugIpOpt to all "false" and then process given element filter
+  std::fill(debugIpOpt, debugIpOpt + DEBUG_IP_TYPE_MAX, false);
+
+  for(auto& itr : _elementsFilter) {
```

Comment:
This if else tree is rather large an can be difficult to support.   Have you thought of refactoring the code to do one of the following:

1. Pass in an array of elements (instead of strings).
2. Create an helper method that will convert an element name to its enumeration type.  

Doing so will pretty much eliminates this if-else tree.

Reply:
What will be the data type of element? Here the "_elementsFilter" array of string is actually passed by the caller of "ReportDebugIpStatus::writeReport" method. Please refer line 1800

void 
ReportDebugIpStatus::writeReport( const xrt_core::device* /*_pDevice*/,
                                  const boost::property_tree::ptree& _pt, 
                                  const std::vector<std::string>& _elementsFilter,
                                  std::ostream & _output) const